lib/OpenQA/WebAPI/Controller/Test.pm

```diff
@@ -77,16 +77,31 @@ sub list {
         };
         push @list, $data;
     }
-    @list = sort { $b->{job}->t_started <=> $a->{job}->t_started || $b->{job}->id <=> $a->{job}->id } @list;
+    @list = sort {
+        if (defined($b->{job}->t_started) && defined($a->{job}->t_started)) {
+            $b->{job}->t_started <=> $a->{job}->t_started || $b->{job}->id <=> $a->{job}->id;
+        }
+        else {
+            $b->{job}->id <=> $a->{job}->id;
+        }
+    } @list;
     $self->stash(running => \@list);
 
     my @scheduled = $self->db->resultset("Jobs")->complex_query(
-        state   => 'scheduled',
+        state   => [OpenQA::Schema::Result::Jobs::PRE_EXECUTION_STATES],
         match   => $match,
         groupid => $groupid,
         assetid => $assetid
     )->all;
-    @scheduled = sort { $b->t_created <=> $a->t_created || $b->id <=> $a->id } @scheduled;
+    # @scheduled = sort { $b->t_created || 'Inf' <=> $a->t_created || 'Inf' || $b->id <=> $a->id } @scheduled;
+    @scheduled = sort {
+        if (defined($b->{job}->t_created) && defined($a->{job}->t_created)) {
```

Comment:
I believe that with the last change of moving ASSIGNED to PRE_EXECUTION_STATES fixed the problems I was having, rendering this change useless :) (And same with the other sorts).